referenceframe/frame_system.go

```diff
@@ -397,6 +402,47 @@ func (sfs *simpleFrameSystem) getFrameToWorldTransform(inputMap map[string][]Inp
 	return srcToWorld, err
 }
 
+// ReplaceFrame transfers replaceMe's children and parentage to replaceWith. ReplaceMe is removed entirely from the frame system.
+// ReplaceWith is not allowed to exist within the frame system at the time of the call.
+func (sfs *simpleFrameSystem) ReplaceFrame(fs FrameSystem, replaceMe, replaceWith Frame) error {
+	if replaceMe.Name() == replaceWith.Name() {
+		return fmt.Errorf("%s (replaceMe) cannot have the same name as %s (replaceWith)", replaceWith.Name(), replaceMe.Name())
+	}
+	if fs.Frame(replaceMe.Name()) == nil {
+		return fmt.Errorf("%s not found in framesystem", replaceMe.Name())
+	}
+	if replaceMe == fs.World() {
+		return fmt.Errorf("%s is the world of the framsystem - which cannot be replaced", replaceMe.Name())
+	}
+	if fs.Frame(replaceWith.Name()) != nil {
+		return fmt.Errorf("%s is not allowed to exist within the framesystem at the time of this call", replaceWith.Name())
+	}
+
+	// get replaceMe's parent
+	replaceMeParent, err := fs.Parent(replaceMe)
+	if err != nil {
+		return err
+	}
+
+	for f, parent := range sfs.parents {
+		// if frame is replaceMe remove it entirely
+		if f == replaceMe {
```

Comment:
I don't think we want this conditional at all.

We can just do
```
			delete(sfs.frames, replaceMe.Name())
			delete(sfs.parents, replaceMe)
```
Before looping through and updating parents